src/index.ts

```diff
@@ -51,29 +57,45 @@ export async function getPort (config?: GetPortInput): Promise<number> {
     portsToCheck.push(memo[memoKey])
   }
 
-  const availablePort = await checkPorts(portsToCheck, options.host)
+  const resolvedPort = await resolvePort(portsToCheck, hostsToCheck)
 
   // Persist
-  await setMemo({ [memoKey]: availablePort }, memoOptions)
+  await setMemo({ [memoKey]: resolvedPort }, memoOptions)
 
-  return availablePort
+  return resolvedPort
 }
 
-async function checkPorts (ports: number[], host: string): Promise<number> {
+async function resolvePort (ports: number[], hosts: string[]): Promise<number> {
+  let availablePort: number[] = []
+
   for (const port of ports) {
-    const r = await checkPort(port, host)
-    if (r) {
-      return r
+    for (const host of hosts) {
+      const _port = availablePort[0] || port
+
+      try {
+        availablePort.push(await checkPort(_port, host))
+      } catch (e) {
+        if (e.code !== 'EADDRINUSE') { // this was expected
```

Comment:
We can move it to `checkPort` `return Promise().catch(err => err.code = EADDRINUSE ? return true : throw err`